### ISSUE TITLE
LSP Integration fails

### ISSUE THREAD

**Issue by Forthoney** [soutaro/steep#1096]
The Steep LSP is failing on VSCode and Neovim, each with a different error message
### Neovim
```
[ERROR][2024-04-05 13:39:03] .../vim/lsp/rpc.lua:734    "rpc"   "/home/castlehoney/.rbenv/shims/steep" "stderr"        "#<NameError: uninitialized constant #<Class:#<Steep::Project::DSL:0x00007f708f3b5d60>>::Bundler>\n  Steepfile:17:in `block (2 levels) in parse'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/project/dsl.rb:203:in `instance_eval'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/project/dsl.rb:203:in `block in target'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/activesupport-7.1.3.2/lib/active_support/tagged_logging.rb:135:in `block in tagged'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/activesupport-7.1.3.2/lib/active_support/tagged_logging.rb:39:in `tagged'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/activesupport-7.1.3.2/lib/active_support/tagged_logging.rb:135:in `tagged'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/project/dsl.rb:202:in `target'\n  Steepfile:3:in `block in parse'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/project/dsl.rb:190:in `instance_eval'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/project/dsl.rb:190:in `block in parse'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/activesupport-7.1.3.2/lib/active_support/tagged_logging.rb:135:in `block in tagged'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/activesupport-7.1.3.2/lib/active_support/tagged_logging.rb:39:in `tagged'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/activesupport-7.1.3.2/lib/active_support/tagged_logging.rb:135:in `tagged'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/project/dsl.rb:189:in `parse'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/drivers/utils/driver_helper.rb:12:in `block in load_config'\n  <internal:kernel>:90:in `tap'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/drivers/utils/driver_helper.rb:11:in `load_config'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/drivers/langserver.rb:36:in `run'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/cli.rb:234:in `process_langserver'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/lib/steep/cli.rb:60:in `run'\n  /home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/steep-1.6.0/exe/steep:11:in `<top (required)>'\n  /home/castlehoney/.rbenv/versions/3.3.0/bin/steep:25:in `load'\n  /home/castlehoney/.rbenv/versions/3.3.0/bin/steep:25:in `<main>'\n"
```

VSCode
```
bundler: failed to load command: steep (/home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/bin/steep)
/home/castlehoney/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/bundler-2.3.8/lib/bundler/definition.rb:481:in `materialize': Could not find steep-1.7.0.dev.3 in any of the sources (Bundler::GemNotFound)
```

**Comment by soutaro** [comment]
Looks like they both are configuration problem.

**Neovim** error looks like it's running `steep` without Bundler. (While there should be a discussion if having implicit dependency to Bundler is okay or not...)

**VSCode** error looks like it's running `steep` without `bundle install`. It's difficult to run Ruby command with correct ruby setup -- rbenv, rvm, anything else?. One workable solution might be using `bin/steep` that is generated by `steep binstub`, and put your environment specific initialization in the file.

**Comment by Forthoney** [comment]
Thank you for the quick response! I retried on Neovim with the following option
```lua
-- old opt
-- cmd = { "steep", "langserver" }
cmd = { "bundle", "exec", "steep", "langserver" }
```
This gives me a different error `[steep] Unexpected error: Unknown name for build_instance: ActiveSupport::TaggedLogging (RuntimeError)`

Maybe the ruby from the LSP does not match use the bundler file from the project? This is an arbitrary guess, but I can't think of any other reason why this would fail. Perhaps someone more knowledgeable on the neovim lsp client would be able to provide more insight.

**Comment by soutaro** [comment]
Is this error reported on an open source project? If it is, we can set up a Google meet or something for troubleshooting?

Does it run from the command line, `bundle exec steep check`?

**Comment by Forthoney** [comment]
@soutaro and I resolved this over a video call. The issue was twofold:

1. The user needs to run `bin/setup` when installing to use steep as a Language Server (silly mistake on my part)
2. The default configuration for steep in the official [neovim-lspconfig](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#steep) may be faulty. Specifically, the default `cmd` value is set to `{"steep", "langserver"}`, but we found that `{"bin/steep", "langserver"}` may actually be the correct setting.

I will also report this to the [neovim-lspconfig](https://github.com/neovim/nvim-lspconfig) repo and hopefully have the instructions changed.

On a side note, a common way people install LSPs and Linters on Neovim is through [Mason](https://github.com/williamboman/mason.nvim). Ruby LSP and Rubocop are already supported on mason, so I think it may be worth putting Steep as well.
